src/Payments.sol

```diff
@@ -1664,6 +1664,70 @@ contract Payments is
 
         return result;
     }
+
+    /**
+     * @notice Gets information about an account - when it would go into debt, total balance, available balance, and lockup rate.
+     * @param token The token address to get account info for.
+     * @param owner The address of the account owner.
+     * @return fundedUntil The epoch at which the account would go into debt given current lockup rate and balance.
+     * @return totalBalance The total balance of the account.
+     * @return availableBalance The available balance (total balance minus current lockup).
+     * @return lockupRate The current lockup rate per epoch.
+     */
+    function getAccountInfo(
+        address token,
+        address owner
+    ) external view returns (
+        uint256 fundedUntil,
+        uint256 totalBalance,
+        uint256 availableBalance,
+        uint256 lockupRate
+    ) {
+        Account storage account = accounts[token][owner];
+        
+        totalBalance = account.funds;
+        lockupRate = account.lockupRate;
+
+        uint256 currentEpoch = block.number;
+        uint256 elapsedTime = currentEpoch - account.lockupLastSettledAt;
+        uint256 simulatedLockupCurrent = account.lockupCurrent;
+
+        if (elapsedTime > 0 && account.lockupRate > 0) {
```

Comment:
@Chaitu-Tatipamula Would be best to have a seperate block for `elapsedTime  <= 0 OR  account.lockupRate > 0==0` and return right away the correct values for that case. There is some complexity here because of trying to understand what happens if this ` if (elapsedTime > 0 && account.lockupRate > 0)` block is NOT hit.